### PR TITLE
Add `Header.of_list` and propose a CoHTTP layer

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

File: lib_cohttp-lwt/multipart_form_cohttp.ml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+let ( <.> ) f g x = f (g x)
+
+module Server = struct
+  let src = Logs.Src.create "multipart_form.cohttp"
+
+  module Log = (val Logs.src_log src : Logs.LOG)
+
+  let handler_exn exn =
+    Log.err (fun m ->
+        m "Got an exception when we tried to parsed a multipart/form: %S"
```
Comment (shonfeder) [inline lib_cohttp-lwt/multipart_form_cohttp.ml:36]:
```suggestion
        m "Got an exception when trying to parse a multipart/form: %S"
```

File: lib_cohttp-lwt/multipart_form_cohttp.mli [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  (** [multipart_form v] returns a well-formed CoHTTP header with a CoHTTP body
+      which can be used by the {i client} implementation of CoHTTP:
+
+      {[
+        let headers, body = Multipart_form_cohttp.Client.multipart_form v in
+        let* resp, body = Client.post ~headers ~body uri in
+        ...
+      ]}
+
+      For more details to construct a {!type:Multipart_form.multipart}, take a
```
Comment (shonfeder) [inline lib_cohttp-lwt/multipart_form_cohttp.mli:13]:
```suggestion
      For more details on constructing a {!type:Multipart_form.multipart}, take a
```

File: lib_cohttp-lwt/multipart_form_cohttp.mli [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      look on the {!module:Multipart_form}. *)
```
Comment (shonfeder) [inline lib_cohttp-lwt/multipart_form_cohttp.mli:14]:
```suggestion
      look at {!module:Multipart_form}. *)
```

**Review by shonfeder — APPROVED** [review]
Looks great to me. Instructive for me to look over :)

I made a few trivial suggestions on documentation phrasing.

Thanks very much for being so responsive to my inquiries.
This addition really has made it much easier for us to drop
the library in for our uses!

**Comment by dinosaure** [comment]
I just tried to use the API by myself and everything works for a simple example. Ready to merge, I will try to cut a release as soon as a I can. Thanks for your feedback!

**Pull request merged**